File: com/white/InvertBinaryTree.java
package com.white;

import java.util.LinkedList;
import java.util.Queue;

public class InvertBinaryTree {

    public TreeNode invertTree(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();

        if (root==null) {
            return null;
        }

        queue.offer(root);

        while(!queue.isEmpty()) {
            int size = queue.size();
            for (int i=0; i<size; i++) {
                TreeNode current = queue.poll();
                TreeNode temp = current.left;
                current.left = current.right;
                current.right = temp;
                if (current.left!=null) {
                    queue.offer(current.left);
                }
                if (current.right!=null) {
                    queue.offer(current.right);
                }
            }
        }

        return root;
    }
}

File: leetcodeCSharp/InvertBinaryTree.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace leetcodeCSharp
{
    public class InvertBinaryTree
    {
        // 226. Invert Binary Tree
        public TreeNode InvertTree(TreeNode root)
        {
            InvertTraverse(root);
            return root;
        }

        private void InvertTraverse(TreeNode node)
        {
            if (node == null)
                return;

            InvertTraverse(node.left);
            InvertTraverse(node.right);

            var temp = node.left;
            node.left = node.right;
            node.right = temp;
        }
    }
}
